packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -77,7 +77,7 @@ class BuildIOSCommand extends BuildSubCommand {
 
     if (!result.success) {
       printError('Encountered error while building for $logTarget.');
-      diagnoseXcodeBuildFailure(result);
+      await diagnoseXcodeBuildFailure(result);
       throwToolExit('');
```

Comment:
Swapped in the message 2 lines earlier. Makes sense to print the definitive implication at the end anyway.